hazelcast/src/hazelcast/client/spi.cpp

```diff
@@ -2207,24 +2231,33 @@ namespace hazelcast {
                     void
                     cluster_view_listener::event_handler::handle_partitionsview(int32_t version,
                                                                                 const std::vector<std::pair<boost::uuids::uuid, std::vector<int>>> &partitions) {
-                        view_listener.client_context_.get_partition_service().handle_event(connection, version, partitions);
+                        auto conn = connection.lock();
+                        view_listener.client_context_.get_partition_service().handle_event(conn, version, partitions);
```

Comment:
`conn` can be empty (nullptr) here. `handle_event` calls `should_be_applied`, and `should_be_applied` dereferences the connection object without a nullptr check at spi.cpp:1693.

Reply:
added null check.